via-router/src/router.rs

```diff
@@ -45,39 +45,25 @@ struct Node<T> {
     route: Vec<MatchCond<T>>,
 }
 
-#[inline(always)]
-fn match_next<'a, T>(
-    bindings: &mut Vec<Binding<'a, T>>,
-    queue: &mut SmallVec<[Branch<'a, '_, T>; 2]>,
-) {
-    if let Some(mut branch) = queue.pop()
-        && let Some(binding) = match branch.segment.take() {
-            Some(segment) => match_next_segment(queue, branch, segment),
-            None => match_trailing_wildcards(branch),
-        }
-    {
-        bindings.push(binding);
-    }
-}
-
-#[inline(always)]
 fn match_next_segment<'a, 'b, T>(
-    queue: &mut SmallVec<[Branch<'a, 'b, T>; 2]>,
-    mut branch: Branch<'a, 'b, T>,
-    segment: (&'b str, [usize; 2]),
+    queue: &mut SmallVec<[Branch<'a, 'b, T>; 1]>,
+    branch: &mut Branch<'a, 'b, T>,
+    (value, range): (&'b str, [usize; 2]),
 ) -> Option<Binding<'a, T>> {
-    let mut results = SmallVec::new();
-
-    let (value, range) = segment;
     let next = branch.path.next();
+    let mut binding = Binding {
+        is_final: next.is_none(),
+        results: SmallVec::new(),
+        range: Some(range),
+    };
 
-    for node in branch.children {
+    for node in branch.children.iter().rev() {
```

Comment:
It's necessary to iterate in reverse since we're call `.pop()` on `SmallVec` instead of next on an iterator.